libs/logs.js

```diff
@@ -16,25 +25,93 @@ const log = {
     });
   },
 
+  writeError: (message, customData = {}) => {
+    log.writeLog('error', message, customData);
+  },
+
   error: (message, requestId, errorCode, customData = {}) => {
     log.log('error', message, requestId, errorCode, customData);
   },
 
+  writeWarn: (message, customData = {}) => {
+    log.writeLog('warn', message, customData);
+  },
+
   warn: (message, requestId, errorCode, customData = {}) => {
     log.log('warn', message, requestId, errorCode, customData);
   },
 
+  writeInfo: (message, customData = {}) => {
+    log.writeLog('info', message, customData);
+  },
+
   info: (message, requestId, errorCode, customData = {}) => {
     log.log('info', message, requestId, errorCode, customData);
   },
 
+  writeVerbose: (message, customData = {}) => {
+    log.writeLog('verbose', message, customData);
+  },
+
   verbose: (message, requestId, errorCode, customData = {}) => {
     log.log('verbose', message, requestId, errorCode, customData);
   },
 
+  writeDebug: (message, customData = {}) => {
+    log.writeLog('debug', message, customData);
+  },
+
   debug: (message, requestId, errorCode, customData = {}) => {
     log.log('debug', message, requestId, errorCode, customData);
   },
+
+  initialize: (event, context) => {
+    requestId = context.awsRequestId;
+    log.writeInfo('Lambda initialize', {
+      source: event.source,
+      detailType: event['detail-type'],
+      path: event.path,
+      method: event.httpMethod,
+      pathParameters: event.pathParameters,
+      queryStringParameters: event.queryStringParameters,
+      userAgent: event.headers?.['User-Agent'],
+    });
+  },
+
+  finalize: (response, error) => {
+    log.writeInfo('Lambda finalize', {
+      statusCode: response?.statusCode ?? 0,
+      error,
+    });
+  },
+
+  wrap: lambda => {
+    return (event, context, callback) => {
```

Comment:
Since we're dealing with promises anyway I think it would be cleaner to rewrite `wrap` to `return async (event, context, callback) => {...}` and utilize `await` instead.

Reply:
Not sure the definition of a Handler should be:
type Handler<TEvent = any, TResult = any> = (event: TEvent, context: Context, callback: Callback<TResult>) => void | Promise<TResult>